src/components/dialog/Dialog.jsx

```diff
@@ -117,10 +116,10 @@ function BaseDialog({
   React.useEffect(() => {
     setDeferredOpen(open);
 
-    if (!supportsTransitions()) {
+    if (!supportsTransitions() || motionPreset === 'none') {
```

Comment:
I think it should be separate variable, you should also handle `onTransitionEnd` in line 288